improver/nbhood/use_nbhood.py

```diff
@@ -402,16 +384,17 @@ def process(self, cube):
         # Loop over any extra dimensions
         cubelist = iris.cube.CubeList([])
         for slice_3d in cube.slices([self.coord_masked, yname, xname]):
-            collapsed_slice = slice_3d.collapsed(
-                self.coord_masked, iris.analysis.MEAN, weights=weights)
+            collapsed_slice = collapsed(slice_3d, self.coord_masked,
+                                        iris.analysis.MEAN, weights=weights)
             cubelist.append(collapsed_slice)
+
         result = cubelist.merge_cube()
         # Promote any scalar coordinates with one point back to dimension
         # coordinates if they were dimensions in the input cube.
         # Take a slice over the coordinate we are collapsing as we do not
         # expect this in the output cube.
         first_slice = next(cube.slices_over([self.coord_masked]))
         result = check_cube_coordinates(first_slice, result)
+        result.remove_coord(self.coord_masked)
```

Comment:
The comment on the line below should be above this line.

Reply:
Done